doc/ondemand.md

```diff
@@ -88,36 +81,42 @@ for (auto tweet : doc["statuses"]) {
 }
 ```
 
-Pros and cons of generic DOM:
-* Straightforward, user-friendly interface (arrays and objects)
-* No lifetime concerns (arrays and objects are often independent of JSON text and parser internal state)
-* Parses and stores everything, using memory/CPU even on values you don't end up using (cost can be brought down some with lazy numbers/strings and top-level iterators)
-* Values stay in memory even if you only use them once
-* Heavy performance drain from [type blindness](#type-blindness).
-
-### SAX (SAJ?) Parsers
-
-The SAX model ("Streaming API for XML") uses streaming to eliminate the high cost of
-parsing and storing the entire JSON. In the SAX model, a core JSON parser parses the JSON document
-piece by piece, but instead of stuffing values in a DOM tree, it passes each value to a callback,
-letting the user use the value and decide for themselves whether to discard it and where to store
-it. or discard it.
-
-This allows users to work with much larger files without running out of memory. Some SAX APIs even
-allow the user to skip values entirely, lightening the parsing burden as well.
-
-The big drawback is complexity: SAX APIs generally have you define a single callback for each type
-(e.g. `string_field(std::string_view key, std::string_view value)`). Because of this, you suffer
-from context blindness: when you find a string you have to check where it is before you know what to
-do with it. Is this string the text of the tweet, the screen name, or something else I don't even
-care about? Are we even in a tweet right now, or is this from some other place in the document
-entirely?
-
-The following is SAX example of the Twitter problem we've seen in the Generic DOM and On Demand
-examples. To make it short enough to use as an example at all, it's heavily redacted: it only solves
-a part of the problem (doesn't get user.screen_name), it has bugs (it doesn't handle sub-objects
-in a tweet at all), and it uses a theoretical, simple SAX API that minimizes ceremony and skips over
-the issue of lazy parsing and number types entirely.
+Pros of the DOM approach:
+* Straightforward, programmer-friendly interface (arrays and objects).
+* Safe: all of the input data has been validated before it is accessed.
+* All of the JSON document is available at once to the programmer.
+
+Cons of the DOM approach:
+* The memory usage scales linearly with the size of the input document.
+* Parses and stores everything, using memory and CPU cycles even on unused values.
+* Performance drain from [type blindness](#type-blindness).
+
+
+What the simdjson library demonstrates is that a DOM API may be quite fast indeed: we can parse files at speeds
+of several gigabytes per second. However, in some instances, it may be possible to achieve even higher speeds.
+
+### Event-Based Parsers (SAX, SAJ, etc.)
+
+
+The event-based model (originally from the "Streaming API for XML") uses streaming to eliminate the  cost of
```

Comment:
double space

Reply:
I am removing all of the double spaces throughout.